mapper/mapper.go

```diff
@@ -2,13 +2,86 @@ package mapper
 
 import (
 	"context"
+	"fmt"
+	"strconv"
 
-	"github.com/ONSdigital/dp-frontend-models/model/homepage"
+	"github.com/ONSdigital/dp-api-clients-go/zebedee"
+	model "github.com/ONSdigital/dp-frontend-models/model/homepage"
+	"github.com/ONSdigital/log.go/log"
 )
 
 // Homepage maps data to our homepage frontend model
-func Homepage(ctx context.Context) homepage.Page {
-	var page homepage.Page
+func Homepage(ctx context.Context, mainFigures []model.MainFigure) model.Page {
+	var page model.Page
 	page.Type = "homepage"
+	page.Metadata.Title = "Home"
+	page.Data.MainFigures = mainFigures
 	return page
 }
+
+// MainFigure maps a single main figure object
+func MainFigure(ctx context.Context, datePeriod string, data zebedee.TimeseriesMainFigure) model.MainFigure {
+	var mf model.MainFigure
+
+	mfData := getDataByPeriod(datePeriod, data)
+	latestData := mfData[(len(mfData) - 1)]
```

Comment:
Not a required set of changes but my personal preference is to store the index value e.g. `latestDataIndex` and `previousDataIndex` then do `mfData[previousDataIndex]`. 

- This would reduce magic numbers; using `len -1` is common for accessing a slice or array as it is a typical 0 index offset but -2 isn't.

Also a personal pref - 'Data' as a variable name could be more meaningful. We have been referring to it as `figure` maybe we can use that. I see later we have a latestFigure and previousFigure which are conversions of the same information stored in latestData and previousFigure - would suggest we just call the data one `latestFigureFloat` and then the string ones `latestFigureStr` - might be a C# thing I picked up at uni but I think this type of naming convention when doing conversions can be helpful on any strongly typed language. This helps us avoid having to be extra creative in naming vars (using synonyms to store the same data but as a different type) but instead able to name the var accurately to the data it is storing.

Reply:
👍